File: Arrays/Algorithms/Searching/Unsorted/self_organizing_search.cpp
#include <bits/stdc++.h>
using namespace std;

int selfOrganizingSearch(vector<int>& arr, int x) {
	for (int i = 0; i < arr.size(); i++) {
		if (arr[i] == x) {
			if (i > 0) swap(arr[i], arr[i - 1]);
			return i;
		}
	}
	return -1;
}

int main() {
	int n, x;
	cin >> n >> x;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	cout << selfOrganizingSearch(arr, x) << endl;
	for (int v : arr) cout << v << ' ';
	cout << endl;
	return 0;
}

File: Arrays/Algorithms/Searching/Unsorted/self_organizing_search.py
def self_organizing_search(arr, x):
	for i in range(len(arr)):
		if arr[i] == x:
			if i > 0:
				arr[i], arr[i - 1] = arr[i - 1], arr[i]
			return i
	return -1

if __name__ == "__main__":
	n, x = map(int, input().split())
	arr = list(map(int, input().split()))
	print(self_organizing_search(arr, x))
	print(*arr)
